worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java

```diff
@@ -128,19 +139,52 @@ public Job processJob(final String jobId) {
         // A mutex lock that all threads for a contract uses while writing into the shared files
         var lock = new ReentrantLock();
```

Comment:
Based on the other comments, it appears we need to communicate the job cancellation "event" from the "main" worker thread to the `PatientClaimsProcessor`'s Futures.

We could use something as simple as this:

- Create an `AtomicBoolean jobStillActive` variable right next to the `lock` and set it to `true`
- `gov.cms.ab2d.worker.adapter.bluebutton.PatientClaimsProcessorImpl#process` will need a new parameter to take it
- Whenever the Worker detects a job cancellation, it should flip the value to `false` in addition to throwing an exception as it does now.
- The first thing `gov.cms.ab2d.worker.adapter.bluebutton.PatientClaimsProcessorImpl#process` should do is check the value of the flag. If `false`, return silently with an empty Future, do nothing else.